ctapipe/tools/stage1.py

```diff
@@ -0,0 +1,868 @@
+
+        cols = {
+            "criteria": self.criteria_names,
+            "counts": self._counts,
+            "cumulative_counts": self._cumulative_counts,
+        }
+        if functions:
+            cols["func"] = self.selection_function_strings
+        return Table(cols)
+
+    def _repr_html_(self):
+        return self.to_table()._repr_html_()
+
+    def __call__(self, value):
+        """
+        Test that value passes all cuts
+
+        Parameters
+        ----------
+        value:
+            the value to pass to each selection function
+
+        Returns
+        -------
+        np.ndarray:
+            array of booleans with results of each selection criterion in order
+        """
+        result = np.array(list(map(lambda f: f(value), self._selectors.values())))
+        self._counts += result.astype(int)
+        self._cumulative_counts += result.cumprod()
+        return result
+
+
+class ImageSelector(Selector):
+    """ for configuring image-wise data checks """
+
+    selection_functions = Dict(
+        help=(
+            "dict of '<cut name>' : lambda function in string format to accept ("
+            "select) a given data value.  E.g. `{'mycut': 'lambda x: x > 3'}` "
+        ),
+        default_value=dict(
+            enough_pixels="lambda im: np.count_nonzero(im) > 2",
+            enough_charge_pe="lambda im: im.sum() > 80",
+        ),
+    ).tag(config=True)
+
+
+def expand_tel_list(tel_list, max_tels, index_map):
+    """
+    un-pack var-length list of tel_ids into 
+    fixed-width bit pattern by tel_index
+
+    TODO: use index_map to index by tel_index rather than tel_id so this can be a
+    shorter array of bools.
+    """
+    pattern = np.zeros(max_tels).astype(bool)
+    pattern[tel_list] = 1
+    return pattern
+
+
+def create_tel_id_to_tel_index_transform(sub):
+    """
+    build a mapping of tel_id back to tel_index:
+    (note this should be part of SubarrayDescription)
+    """
+    idx = np.zeros(max(sub.tel_indices) + 1)
+    for key, val in sub.tel_indices.items():
+        idx[key] = val
+
+    # the final transform then needs the mapping and the number of telescopes
+    return partial(expand_tel_list, max_tels=len(sub.tel) + 1, index_map=idx)
+
+
+class Stage1ProcessorTool(Tool):
+    name = "ctapipe-stage1-process"
+    description = __doc__ + f" This currently writes {DL1_DATA_MODEL_VERSION} DL1 data"
+    examples = """
+    To process data with all default values:
+    > ctapipe-stage1-process --input events.simtel.gz --output events.dl1.h5 --progress
+    
+    Or use an external configuration file, where you can specify all options:
+    > ctapipe-stage1-process --config stage1_config.json --progress 
+    
+    The config file should be in JSON or python format (see traitlets docs). For an 
+    example, see ctapipe/examples/stage1_config.json in the main code repo. 
+    """
+
+    output_filename = Unicode(
+        help="DL1 output filename", default_value="events.dl1.h5"
```

Comment:
pep8: `help="` -> `help = "`
Same below

Reply:
No, pep8 says no space for around = for keyword arguments.